### DISCUSSION TITLE
Latest mini hacks with WC3 - can't connect on wifi

### DISCUSSION THREAD

**Discussion by teredactle** [gtxaspec/wz_mini_hacks#612] [Q&A]
I've loaded up the latest mini hacks to test out go2rtc and see if this, in combination with the wc3 beta RTSP fw will work. So it doesn't, same issues as running the regular RTSP server.

However I've run into the problem that the camera won't connect over WiFi; my testing has been done with POE via the UCTRONICS dongle, which works fine, camera boots and I can connect to its IP, hard wired.

If I just power up the camera with a microUSB cable, it powers up but never connects to WIFI unless I take out the SD card with mini hacks; then it boots up and registers to WIFI just fine.

Is this normal behaviour? I assumed that once it's registered to WIFI with the Wyze fw, using mini hacks won't affect it registering to WIFI...

**Comment by virmaior** [comment]
wz_mini_hacks should be using the same basis for connnecting.
the only thing I can imagine is that some of the more recent version of wyze's firmware encrypt the wifi password but those versions are all past the ones you should be using for wz_mini_hacks

**Reply by teredactle** [reply]
I'm having the same issue with both the recommended version of fw, as well as with the wyze rtsp beta fw. Works great with the poe dongle, but just powered up with a cable won't connect to wifi. If i remove sd card, bypassing mini hacks, there is no problem.

**Reply by virmaior** [reply]
what version of wyze firmware are you running?

**Reply by teredactle** [reply]
4.61.0.3, but also tried 4.36.9.139 and had the same exact issue.

**Reply by Bob-AOL** [reply]
Once you set the camera to receive an IP address via the UCTRONICS adapter in the wz_mini.conf file the default camera wi-fi is disabled by wz_mini. The USB/ETH connection in wz_mini.conf will either need to be disabled (you can continue to use it for POE) or the SD card removed prior to booting for wi-fi to be reactivated in the camera firmware.

**Reply by teredactle** [reply]
Uhmm, that sucks... is there a way around that? What if I set Bonding_Enabled to true?  
From how I interpret this function, this is the fix, and switch the primary interface to wlan0 and secondary to eth0. I would like to use the cam mostly with wifi for when I travel (baby cam) but when I'm home have the option to use the UCTRONICS adapter on a POE drop.

##### NETWORK INTERFACE BONDING #####
BONDING_ENABLED="false"
BONDING_PRIMARY_INTERFACE="eth0"
BONDING_SECONDARY_INTERFACE="wlan0"
BONDING_LINK_MONITORING_FREQ_MS="100"
BONDING_DOWN_DELAY_MS="5000"
BONDING_UP_DELAY_MS="5000"
BONDING_FAIL_OVER_MAC="0"

**Reply by Bob-AOL** [reply]
I've wondered about that but never tried it.

The issue is which interface is initiating dhcp discover to obtain the IP address -- and how that's determined. The failover MAC when bonding maybe? Dunno.

The easiest way is editing the wz_mini.conf file outside the camera for either eth or not and rebooting with the new setting accordingly. Once the UCTRONICS adapter has been recognized on the network it will be plug and play thereafter -- especially easy if there's an IP address reservation for the MAC address. -- and applies primarily to your home network where the UCTRONICS adapter resides.

**Reply by Bob-AOL** [reply]
How about just using 2 SD cards -- a 'home' and an 'away' version for each connection type?

**Reply by teredactle** [reply]
That's a no go, makes it more complicated then I'd like, well too bad. I also re-read the section on this bonding and I don't quite get the reference to the wireless interface, how it's supposed to work, at the end of the section it says this "Currenly supported with ethernet adapters and usb-direct mode".... not very clear on the wireless part....

**Reply by Bob-AOL** [reply]
What's complicated about switching SD cards and rebooting the camera? One for wi-fi and one for hardwired. Doesn't get much more simple than that.

Or just get another camera. One for the road and one that stays home.

**Reply by gtxaspec** [reply]
It means that bonding works with wired Ethernet adapters or USB direct networking interface methods.

I suppose for your use case you could try to use bonding, it should work

**Reply by Bob-AOL** [reply]
Not exactly a failover situation if the primary interface would need to be previously active to actually initiate the switch upon failure for the secondary interface to request an IP address to come online...., unless it boots to whatever it sees pending the results of the bonding configuration being initiated without an actual failover condition being detected.

Easy enough to, oh..., I dunno...., test? LOL

**Reply by bjs-pdx** [reply]
You can set which interface is primary in the bonding section of the config file:
```
##### NETWORK INTERFACE BONDING #####
BONDING_ENABLED="false"
BONDING_PRIMARY_INTERFACE="eth0"
BONDING_SECONDARY_INTERFACE="wlan0"
BONDING_LINK_MONITORING_FREQ_MS="100"
BONDING_DOWN_DELAY_MS="5000"
BONDING_UP_DELAY_MS="5000"
BONDING_FAIL_OVER_MAC="0"

```

**Reply by bjs-pdx** [reply]
I have one of my V3 cameras running using bonding and NO ethernet device connected.  It is using the wifi module and seems to be working correctly even with reboots.

![Screenshot 2023-10-04 at 10-38-25 T31(WYZE_CAKP2JFUS) on 4 36 9 139 running wz_mini Wed Sep 20 04 17 50 PM PDT 2023 as Backyard](https://github.com/gtxaspec/wz_mini_hacks/assets/14082937/b97c5e18-598c-453d-a56e-48cb9035ecdd)

**Reply by Bob-AOL** [reply]
Will it revert to eth0 if there's no wi-fi?

**Reply by bjs-pdx** [reply]
Only if I hookup an ethernet adapter.

**Reply by Bob-AOL** [reply]
So it knows what's available (or not available) on boot and chooses the correct option either way aside from it's basic failover function to the secondary interface if the primary goes down -- (provided the primary isn't a POE connection and the sole powering source for the camera, of course.)

In other words, it does what you'd think it should do.

**Reply by bjs-pdx** [reply]
Yes, that's correct.  I have the camera powered by a USB cable so no POE at the moment.  If I connect a POE ethernet it should select that at boot.  It isn't in a convenient location to test that at the moment as I don't have ethernet cable running to that location.  May do that later as I was just trying to get the camera working quickly to cover my backyard solar panels.

**Reply by Bob-AOL** [reply]
Very cool. The info/confirmation is appreciated.

It would then seem like @teredactle is missing something else in the config for his desired solution.

**Comment by bjs-pdx** [comment]
> I assumed that once it's registered to WIFI with the Wyze fw, using mini hacks won't affect it registering to WIFI...

If you enable POE in the config file WIFI is disabled, unless you also turn on bonding.  That is why WiFi works if you pull the SD card.

**Reply by Bob-AOL** [reply]
POE has no effect on wi-fi and in fact works wonderfully to power wi-fi v3s. Enabling _ETH_ disables wi-fi _IF_ a USB/ETH adapter is successfully set up to generate dhcp discover either using wi-fi to do so or already successfully set up previously to obtain an IP address for the ethernet hardware. That's what over rides wi-fi as no IP address is then active for it.

If using an ethernet adapter that has not yet been successfully set up with the camera,  wi-fi will still prevail regardless of what the file is configured to do.

I still don't get the bonding.

**Reply by bjs-pdx** [reply]
Sorry, I meant ETH.  I conflate the two.  Bonding would be useful if you are connected via cable and might lose that connection.  More likely to lose WiFi.

**Reply by teredactle** [reply]
I've tested this and camera never hook up to wifi, bonding on or bonding off. I'd like to know if someone has this working though and the steps to get this working, without taking the SD card out. Not clear on the bonding option.

@Bob-AOL I'd rather not have to juggle 2 SD cards, changes are high I will loose one :)

**Reply by Bob-AOL** [reply]
Well, you have two choices if the bonding is not working for your particular purpose.

1. Use the two SD card approach which is by far the simplest solution.

2. Edit the wz_mini.conf file on the existing SD card via SSH or webUI to the desired connectivity type for your NEXT location _each time_ you anticipate changing locations -- which will obviously disable the existing connectivity type for the current location (unless both connection types are available to toggle back and forth if necessary.)

So are you saying you can't enable wi-fi at all by any of the suggested methods?

**Comment by bjs-pdx** [comment]
> I've tested this and camera never hook up to wifi, bonding on or bonding off. I'd like to know if someone has this working

Try disabling eth in the config file.  Set ENABLE_USB_ETH="false" and see if you connect to wifi.

```
#####NETWORKING#####
CUSTOM_HOSTNAME="Backyard"

ENABLE_USB_ETH="true"
ENABLE_USB_ETH_MODULE_AUTODETECT="true"
ENABLE_USB_ETH_MODULE_MANUAL=""
USB_ETH_MAC_ADDR=""

```

**Reply by Bob-AOL** [reply]
Same as not using mini_hacks at all. Pull SD card. Camera defaults to wi-fi. The end..., unless there's a whole other network issue affecting the  camera's ability to even send a discover packet even though wi-fi is trying to..., or the firmware flash of the existing version got hosed somehow.

Start from scratch with a factory reset, fresh SD card and the recommended 139xxxx firmware and follow the _**Wyze**_ directions for re-registering the camera doing everything with the Wyze app.

THEN try wx_mini again. Anything less is just pissing up a rope at this point.

**Comment by teredactle** [comment]
Once I get some time I will give it a go again, likely not for a while, but will come back here, re-read, redo :)

**Comment by archandanime** [comment]
I had same issue when upgrading to 4.36.11.5859 and 4.36.10.4054, then I had to rollback to 4.36.9.139 again. The it's the stock firmware that caused the issue because I upgraded from a working 4.36.9.139 with Wyze mobile app.
